src/service/action-impl.js

```diff
@@ -536,27 +536,26 @@ export class ActionService {
     if (tagName == 'INPUT' || tagName == 'SELECT') {
       const type = target.getAttribute('type');
 
-      // TODO(blueyedgeek, #10729): Provide valueAsNumber instead of casting.
-      detail['value'] = (type == 'range') ? Number(target.value) : target.value;
-
-      // Expose HTMLInputElement properties based on type.
-      if (tagName == 'INPUT') {
-        if (type == 'checkbox' || type == 'radio') {
-          detail['checked'] = target.checked;
-        } else if (type == 'range') {
-          // TODO(choumx): Perhaps we shouldn't cast since min/max is also
-          // available on input[type="date|time|number"].
-          detail['min'] = Number(target.min);
-          detail['max'] = Number(target.max);
-        }
+      detail['value'] = target.value;
+
+      detail['valueAsNumber'] = (target.valueAsNumber !== undefined)
+          ? target.valueAsNumber
```

Comment:
Is there some conversion that's different? Can't we just always use the coerced `target.value` for consistency?

Reply:
Done.